### PR TITLE
add: page title

### PR DESCRIPTION
**Pull request by Qsnh**
Feature

Changes proposed in this pull request:
- page title

### CODE REVIEW

File: src/views/Strategy.vue [hunk 1]
```diff
@@ -1,14 +1,19 @@
...
+import { setPageTitle } from '@/helpers/utils';
 
 const route = useRoute();
 const { strategies } = useApp();
 const { minifiedStrategiesArray } = useSearchFilters();
 
 const strategy = computed(() => strategies.value[route.params.name]);
+
+onMounted(() => {
+  setPageTitle('page.title.strategy');
```
Comment (samuveth) [inline src/views/Strategy.vue:15]: Looks good!
Here we could also use the `strategy.key` in the title

File: src/views/SpaceProposals.vue [hunk 2]
```diff
@@ -49,7 +50,10 @@ async function loadProposals(skip = 0) {
   store.space.proposals = store.space.proposals.concat(proposalsObj);
 }
 
-onMounted(load());
+onMounted(() => {
+  setPageTitle('page.title.space.proposals', { space: props.space.name });
```
Comment (samuveth) [inline src/views/SpaceProposals.vue:54]: It shows `Proposals - ENS - Snapshot` as title in spaces. I think just `ENS Proposals - Snapshot` would be better.

More space title examples:
`About ENS - Snapshot`
`Create ENS proposal - Snapshot`

@bonustrack what are your thoughts?

**Review by mktcode — APPROVED** [review]
Looks good. Great work @Qsnh! :) Can be merged as is.

I support @samuveth suggestion for the title pattern though and I'm not 100% sure if this `utils.ts` is the "correct" place for the `setPageTitle` function. These helpers do just some calculation/formatting but changing page titles is "busyness logic" so to say and will potentially be enriched with other meta data improvements (proper sharing features etc.). I'd maybe put this in a new `useMeta` composable. `onMounted` could be moved there too, so there'd be a bit less code in those components that need it only for the page title.

File: src/views/Strategy.vue (hunk 1 above)
Comment (Qsnh) [reply src/views/Strategy.vue:15]: 👌🏻

File: src/views/SpaceProposals.vue (hunk 2 above)
Comment (Qsnh) [reply src/views/SpaceProposals.vue:66]: okay, i will review ite.

**Comment by Qsnh** [comment]
> Looks good. Great work @Qsnh! :) Can be merged as is. (conflicts resolved)
> 
> I support @samuveth suggestion for the title pattern though and I'm not 100% sure if this `utils.ts` is the "correct" place for the `setPageTitle` function. These helpers do just some calculation/formatting but changing page titles is "busyness logic" so to say and will potentially be enriched with other meta data improvements (proper sharing features etc.). I'd maybe put this in a new `useMeta` composable. `onMounted` could be moved there too, so there'd be a bit less code in those components that need it only for the page title.

sounds goods.let's me optimize the code.

File: src/locales/default.json [hunk 3]
```diff
@@ -375,5 +375,23 @@
     "show": "Show",
     "reply": "Reply",
     "load_more": "Load more"
+  },
+  "page": {
+    "title": {
+      "home": "Home",
```
Comment (bonustrack) [inline src/locales/default.json:381]: Home should use the title just "Snapshot"
```suggestion
      "home": "Snapshot",
```

File: src/helpers/utils.ts [hunk 4]
```diff
@@ -142,3 +143,8 @@ export function calcToSeconds(value, unit) {
   if (unit === 'h') return value * 60 * 60;
   if (unit === 'd') return value * 60 * 60 * 24;
 }
+
+export function setPageTitle(message, params = {}) {
+  const { t } = i18n.global;
+  document.title = t(message, params) + ' - Snapshot';
```
Comment (bonustrack) [inline src/helpers/utils.ts:149]: I would remove this here
```suggestion
  document.title = t(message, params);
```

**Pull request merged**